modules/context/api.go

```diff
@@ -413,7 +413,7 @@ func RepoRefForAPI(next http.Handler) http.Handler {
 				return
 			}
 			ctx.Repo.CommitID = ctx.Repo.Commit.ID.String()
-		} else if len(refName) == 40 {
+		} else if len(refName) >= 7 && len(refName) <= 40 {
```

Comment:
I doubt that that will improve the situation.
"Great. Now I cannot even get the README.md".
Maybe it would be better to check whether that file is tracked when the if three lines below is entered before returning the error.

Or do I misunderstand what this is supposed to do?
Regarding the else block below, I am fairly certain that I am misunderstanding something here.

Reply:
I don't think hard-coding the length is a good idea. Maybe git will use SHA256 in future.

https://stackoverflow.com/questions/28159071/why-doesnt-git-use-more-modern-sha

Reply:
@delvh no this won't happen, as we check if such a commit exists. If not, we fall back to treating the entire thing as a TreePath and return Repo.DefaultBranch as ref
@wxiaoguang this is all over the codebase, definitily a different issue.

Reply:
Maybe we can introduce a function like `IsPossibleGitCommit` and use it from now on. It's easier to understand and maintain.